playground/src/components/contract/components/deployContractDialog.tsx

```diff
@@ -98,12 +98,17 @@ export function DeployContractDialog({
                 <Select
                   value={initializer?.name ?? ''}
                   label="Initializer"
-                  disabled={!contractArtifact.functions.some(fn => fn.isInitializer)}
+                  disabled={
+                    !contractArtifact.nonDispatchPublicFunctions
+                      .concat(contractArtifact?.functions ?? [])
+                      .some(fn => fn.isInitializer)
+                  }
                   onChange={e => {
                     setInitializer(getInitializer(contractArtifact, e.target.value));
                   }}
                 >
-                  {contractArtifact.functions
+                  {contractArtifact.nonDispatchPublicFunctions
+                    .concat(contractArtifact?.functions ?? [])
```

Comment:
(Oops, should have commented here!)
These are different types - I added the fn `getAllFunctionAbis` for this reason (so we can check all fns for something at once). If that's ok in this context, ignore this!

Reply:
Oh, didn't know the helper existed! Changed the code to use it since it's more elegant 😁